src/janus-controller/service/service.go

```diff
@@ -176,28 +181,38 @@ func (s *Service) RunApi(port string) {
 		}
 
 		go func() {
-			helper.TryUntilNoError(device.Client.Status, 600) //check if agent is already up and running
-			log.Println("\nChanging invitation for agent ", ip)
+			helper.TryUntilNoError(device.Client.Status, 30) //check if agent is already up and running
 
-			invitationID, _, _ := agents.ChangeInvitations(s.ServerClient, device.Client)
+			log.InfoLogger("Agent %s: Exchanging invitation...", ip)
+
+			invitationID, _, err := agents.ChangeInvitations(s.ServerClient, device.Client)
 			device.ConnectionID = invitationID
+			if err != nil {
+				log.ErrorLogger("Agent %s: %s", ip, err)
+
+				return
+			}
+
+			log.InfoLogger("Agent %s: Invitation accepted with ID %s", ip, invitationID)
 
 			time.Sleep(5 * time.Second)
 
+			log.InfoLogger("Agent %s: Getting credential...", ip)
 			cred, err := agents.GetCredential(device.Client, "cred_def_id", s.CredDefinitionId) //issue new credential only if no previous created
 			if err != nil && err.Error() == "empty" {
-				log.Println("\nIssuing credential for agent ", ip)
+				log.InfoLogger("Agent %s: Issuing credential", ip)
+
 				agents.IssueCredential(s.ServerClient, s.CredDefinitionId, device.ConnectionID, permissionList)
 				cred, err = helper.TryUntilNoError(func() (acapy.Credential, error) {
 					return agents.GetCredential(device.Client, "cred_def_id", s.CredDefinitionId)
 				}, 20)
 
 				if err != nil {
-					log.Println("Timeout on agents.GetCredential")
+					log.InfoLogger("Timeout on agents.GetCredential")
```

Comment:
This should be a error log, even not using the err variable content
```suggestion
					log.ErrorLogger("Agent %s: %s", ip, "Timeout getting a credential")
```